lib/tasks/readyset.rake

```diff
@@ -1,13 +1,87 @@
 require 'colorize'
 require 'erb'
 require 'progressbar'
+require 'terminal-table'
 
 namespace :readyset do
+  desc 'Creates a cache from the given query ID'
+  task :create_cache, [:id] => :environment do |_, args|
+    if args.key?(:id)
+      Readyset.create_cache!(id: args[:id])
+    else
+      puts 'A query ID must be passed to this task'
+    end
+  end
+
+  desc 'Creates a cache from the given query ID whose queries will never fall back to the ' \
+    'primary database'
+  task :create_cache_always, [:id] => :environment do |_, args|
+    if args.key?(:id)
+      Readyset.create_cache!(id: args[:id], always: true)
+    else
+      puts 'A query ID must be passed to this task'
+    end
+  end
+
+  desc 'Prints a list of all the queries that ReadySet has proxied'
+  task proxied_queries: :environment do
+    rows = Readyset::Query::ProxiedQuery.all.map do |q|
+      [q.id, q.text, q.supported, q.count]
+    end
+    table = Terminal::Table.new(headings: [:id, :text, :supported, :count], rows: rows)
+
+    puts table
+  end
+
+  namespace :proxied_queries do
+    desc 'Creates caches for all of the supported queries on ReadySet'
+    task cache_all_supported: :environment do
+      Readyset::Query::ProxiedQuery.cache_all_supported!
+    end
+
+    desc 'Clears the list of proxied queries on ReadySet'
+    task drop_all: :environment do
+      Readyset.raw_query('DROP ALL PROXIED QUERIES')
+    end
+
+    desc 'Prints a list of all the queries that ReadySet has proxied that can be cached'
+    task supported: :environment do
+      rows = Readyset::Query::ProxiedQuery.all.
+        select { |query| query.supported == :yes }.
+        map { |q| [q.id, q.text, q.count] }
+      table = Terminal::Table.new(headings: [:id, :text, :count], rows: rows)
+
+      puts table
+    end
+  end
+
+  desc 'Prints a list of all the cached queries on ReadySet'
+  task caches: :environment do
+    rows = Readyset::Query::CachedQuery.all.map do |q|
+      [q.id, q.name, q.text, q.always, q.count]
+    end
+    table = Terminal::Table.new(headings: [:id, :name, :text, :always, :count], rows: rows)
+
+    puts table
+  end
+
   namespace :caches do
+    desc 'Drops the cache with the given name'
+    task :drop, [:name] => :environment do |_, args|
+      if args.key?(:name)
+        Readyset.drop_cache!(args[:name])
+      else
+        puts 'A cache name must be passed to this task'
+      end
+    end
+
+    desc 'Drops all the caches on ReadySet'
+    task drop_all: :environment do
+      Readyset::Query::CachedQuery.drop_all!
+    end
```

Comment:
**_suggestion:_** Do you think these tasks would be considered 'dangerous' despite being copies of data?

Potential safety measure: adding a confirmation prompt and comparing it against a '`confirm_token`'. [Reference article with some neat ideas.](https://www.lynnbright.com/6-tips-to-ensure-your-rake-task-runs-smoothly/)

Reply:
I think that's a great idea! But I will probably leave it as-is for now in the interest of time